app/src/main/java/com/asfoundation/wallet/wallet_validation/ValidationSuccessPresenter.kt

```diff
@@ -0,0 +1,26 @@
+package com.asfoundation.wallet.wallet_validation
+
+import com.asfoundation.wallet.poa.ProofOfAttentionService
+import io.reactivex.disposables.CompositeDisposable
+
+class ValidationSuccessPresenter(
+    private val view: ValidationSuccessFragment,
+    private val service: ProofOfAttentionService
+) {
+
+  private val disposables: CompositeDisposable = CompositeDisposable()
+
+  fun present() {
+    view.setupUI()
+  }
+
+  fun updatePoA() {
+    service.setWalletValidated()
```

Comment:
This doesn’t go according to our MVP architecture (view doesn’t need to know about presenter methods). What I would do in this case is create a subject in the view, and have the presenter listen to a view method that returns it and react on it.